packages/@magic-sdk/provider/src/modules/base-module.ts

```diff
@@ -26,6 +26,18 @@ export class BaseModule {
    * Emits promisified requests to the Magic `<iframe>` context.
    */
   protected request<ResultType = any, Events extends EventsDefinition = void>(payload: Partial<JsonRpcRequestPayload>) {
+    if (
+      this.sdk.thirdPartyWallets.isConnected &&
```

Comment:
Question, are we aiming to route all eth payloads and auth payloads to 3p wallets?

Reply:
https://github.com/magiclabs/magic-js/pull/742/files#diff-43ccfa93c9199f6798e8538eb4ed018fb90cb2aae461a49d0bed91fb5e862994R17

From this line, it seems there is only a specific list of payloads that need to be routed.

Reply:
Yes once a user is connected to a third party wallet, we route all requests to that wallet. We have specific handling for certain reqs (which you linked). But all others, including any web3 rpc requests we route to the 3pw

Reply:
Would there be cases when developer calls loginWithSMS with 3pw connected, And the call might just hang?